torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -472,12 +472,16 @@ ProcessGroupNCCL::WorkNCCL::WorkNCCL(const WorkNCCL& w)
 ProcessGroupNCCL::WorkNCCL::~WorkNCCL() = default;
 
 bool ProcessGroupNCCL::WorkNCCL::isCompleted() {
-  checkAndSetException();
+  if (!ncclComm_->isAborted()) {
+    checkAndSetException();
+  }
```

Comment:
I guess one question I have is what would be the behavior in the `isAborted() == true` path.

Reply:
yea, do we need to early-exit the watchdog loop in that case to avoid touching any other apis? or is it fine to just keep the loop running

Reply:
It seems to be ok to continue as is, although it's definitely not very safe. We'll continue to run things like "finishedGPUExecutionInternal" which queries the cuda event that is outside nccl so should still be safe.

Reply:
yeah, `isCompleted()` will return true. I guess that might be okay even though `isCompleted()` is pybinded to a user-facing API: `work.is_completed()`. As like, the collective is indeed completed, just unsuccessfully.